python/common/org/python/types/Bool.java

```diff
@@ -65,41 +65,135 @@ public Bool(long int_val) {
         __doc__ = ""
     )
     public org.python.Object __lt__(org.python.Object other) {
+        if (other instanceof org.python.types.Int) {
+            return new org.python.types.Bool( (((org.python.types.Bool) this).value ? 1 : 0) < ((org.python.types.Int) other).value);
+        } else if (other instanceof org.python.types.Bool) {
+            return new org.python.types.Bool( (((org.python.types.Bool) this).value ? 1 : 0) < (((org.python.types.Bool) other).value ? 1 : 0));
+        } else if (other instanceof org.python.types.Float) {
+            return new org.python.types.Bool( (((org.python.types.Bool) this).value ? 1.0 : 0.0) < (((org.python.types.Float) other).value));
+        } else if (other instanceof org.python.types.Str) {
+            throw new org.python.exceptions.TypeError("unorderable types: bool() < " + other.typeName() + "()");
+        } else if (other instanceof org.python.types.Dict) {
+            throw new org.python.exceptions.TypeError("unorderable types: bool() < " + other.typeName() + "()");
+        } else if (other instanceof org.python.types.NoneType) {
+            throw new org.python.exceptions.TypeError("unorderable types: bool() < " + other.typeName() + "()");
+        } else if (other instanceof org.python.types.Set) {
+            throw new org.python.exceptions.TypeError("unorderable types: bool() < " + other.typeName() + "()");
+        }
         throw new org.python.exceptions.NotImplementedError("bool.__lt__() has not been implemented.");
```

Comment:
What valid comparisons remain to be implemented? Is there any reason that the last four comparisons (Str, Dict, None, Set) could be collapsed into a single "else - unorderable type"?